atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/LightweightOppToken.java

```diff
@@ -34,6 +37,35 @@ public LightweightOppToken(byte[] bytes) {
         this.bytes = bytes;
     }
 
+    public static LightweightOppToken fromHex(String token) {
+        // OPP tokens should be lowercase already, use upper if needed, and convert mixed to uppercase if needed
+        byte[] bytes = isAllLowercaseOrDigits(token)
+                ? BaseEncoding.base16().lowerCase().decode(token)
+                : isAllUppercaseOrDigits(token)
+                        ? BaseEncoding.base16().decode(token)
+                        : BaseEncoding.base16().decode(token.toUpperCase(Locale.ROOT));
```

Comment:
Looks good as is.

<ideation with little bearing on the result follows…> We could probably simplify out the isAllUpperCase check in favor of toUpperCase which is effectively a no-op if the value is already upper, and no different in the fallback case.